src/olympia/devhub/tests/test_views_versions.py

```diff
@@ -658,26 +657,81 @@ def test_version_history_mixed_channels(self):
         )
 
     def test_pending_activity_count(self):
+        v1 = self.addon.current_version
+        v1.update(created=self.days_ago(1))
         v2, _ = self._extra_version_and_file(amo.STATUS_AWAITING_REVIEW)
+        v3, _ = self._extra_version_and_file(amo.STATUS_APPROVED)
         # Add some activity log messages
+        ActivityLog.objects.create(
+            amo.LOG.REVIEWER_REPLY_VERSION, v1.addon, v1, user=self.user
+        )
         ActivityLog.objects.create(
             amo.LOG.REVIEWER_REPLY_VERSION, v2.addon, v2, user=self.user
         )
         ActivityLog.objects.create(
             amo.LOG.REVIEWER_REPLY_VERSION, v2.addon, v2, user=self.user
         )
 
-        response = self.client.get(self.url)
+        with self.assertNumQueries(43):
```

Comment:
This test is to prevent additional queries being added to the view? Instead of documenting here could you put this in a ticket and link the ticket in the code?

Reply:
This is a test to prove my own changes don't scale badly and make the page performance any worse (note that there aren't extra queries caused by the number of versions or number of activities on those versions). I'll file an issue about the various bits we could improve on that I discovered along the way (though it's very low priority/unlikely to be worked on any time soon, it has been that way forever)

Reply:
Still better to track in an issue.

Reply:
https://github.com/mozilla/addons/issues/15037